include/boost/geometry/algorithms/detail/covered_by/implementation.hpp

```diff
@@ -57,34 +60,26 @@ struct use_relate
     }
 };
 
-}} // namespace detail::covered_by
-#endif // DOXYGEN_NO_DETAIL
-
-#ifndef DOXYGEN_NO_DISPATCH
-namespace dispatch
-{
-
-template <typename Point, typename Box>
-struct covered_by<Point, Box, point_tag, box_tag>
+struct geometry_covered_by_box
 {
-    template <typename Strategy>
-    static inline bool apply(Point const& point, Box const& box, Strategy const& strategy)
+    template <typename Geometry, typename Box, typename Strategy>
+    static inline bool apply(Geometry const& geometry, Box const& box, Strategy const& strategy)
     {
-        return strategy.covered_by(point, box).apply(point, box);
-    }
-};
+        using point_type = typename point_type<Geometry>::type;
+        using box_type = model::box<point_type>;
 
-template <typename Box1, typename Box2>
-struct covered_by<Box1, Box2, box_tag, box_tag>
-{
-    template <typename Strategy>
-    static inline bool apply(Box1 const& box1, Box2 const& box2, Strategy const& strategy)
-    {
-        assert_dimension_equal<Box1, Box2>();
-        return strategy.covered_by(box1, box2).apply(box1, box2);
+        box_type box_areal;
+        geometry::envelope(geometry, box_areal, strategy);
+        return strategy.covered_by(box_areal, box).apply(box_areal, box);
```

Comment:
This approach while giving the correct result is suboptimal because it cannot be interrupted  if e.g. one of the points/segments of a geometry is outside the box. The whole envelope is calculated each time. If that's acceptable then ok. But if it isn't I'd at least put here big TODO.